opm/parser/eclipse/EclipseState/IOConfig/RestartConfig.cpp

```diff
@@ -155,19 +256,86 @@ namespace Opm {
 
         if( items.size() <= RESTART_index ) return {};
 
-        return restart( std::stoul( items[ RESTART_index ] ) );
+        return RestartSchedule( std::stoul( items[ RESTART_index ] ) );
+    }
+
+
+    void RestartConfig::update( size_t step, const RestartSchedule& rs) {
+        if (step == 0)
+            this->restart_schedule->updateInitial( rs );
+        else
+            this->restart_schedule->update( step, rs );
     }
 
-    DynamicState< RestartConfig::restart > RestartConfig::rstconf(
-            const SCHEDULESection& schedule,
-            std::shared_ptr< const TimeMap > timemap ) {
+
+
+
+    /*
+      The time related mnemonics BASIC= and FREQ= have been stripped
+      out before calling this method, the other keywords just come
+      verbatim from the deck; in particular we do not split KEY=int
+      keywords - there are some of them.
+    */
+
+    void RestartConfig::addKeywords( size_t step, const std::vector<std::string>& keywords) {
+        if (keywords.size() > 0) {
+            std::set<std::string> kw_set = this->restart_keywords->back();
+            for (const auto& kw : keywords )
+                insertKeyword( kw_set , kw );
+
+            if (step == 0)
+                this->restart_keywords->updateInitial( kw_set );
+            else
+                this->restart_keywords->update( step , kw_set );
+        }
+    }
+
+
+    /*
+      When handling the RPTRST keyword based on integer controls we
+      can actually remove keywords from the write set by passing in a
+      control value of 0. With our implementation it is not possible
+      to remove keywords with the string based menonics.
+    */
+
+    void RestartConfig::updateKeywords( size_t step, const std::vector<int>& integer_controls) {
+        std::set<std::string> keywords = this->restart_keywords->back();
+
+        for (size_t index = 0; index < integer_controls.size(); index++) {
+            if (index == 26) {
+                if (integer_controls[index] > 0) {
+                    insertKeyword( keywords , "PCOW" );
```

Comment:
This is confusing. If POCW+PCOG hasn't been added because integer_controls[26] (can be done outside of the loop by the way, clearer that way) <= 0, why would you need to erase them?